src/Storages/StorageS3.cpp

```diff
@@ -660,8 +660,18 @@ Pipe StorageS3::read(
     Block block_for_format;
     if (isColumnOriented())
     {
+        auto fetch_columns = column_names;
+        const auto & virtuals = getVirtuals();
```

Comment:
Does it mean that virtual columns shadow ordinary columns?
But it's better to have opposite: if a file has `_path` column, then this column will be not virtual, and the virtual `_path` column will be unaccessible.

Reply:
Done